client/src/components/pages/MembersPage/components/CreateForm/CreateForm.js

```diff
@@ -0,0 +1,186 @@
+                Administrator
+              </label>
+            </CheckboxRow>
+            <CheckboxRow>
+              <label htmlFor="musician">
+                <input
+                  type="checkbox"
+                  id="musician"
+                  name="musician"
+                  value="musician"
+                />
+                Musician
+              </label>
+            </CheckboxRow>
+          </Row>
+        </Column>
+      </Row>
+      <Row>
+        <InputField
+          type="text"
+          name="newInstrument"
+          placeholder="new instrument"
+          ref={newInstrumentInputRef}
+        />
+        <AddNewMembersInstrument
+          onClick={event => {
+            event.preventDefault()
+            newInstrumentInputRef.current.value &&
+              addNewMembersInstrument(newInstrumentInputRef.current.value)
+            newInstrumentInputRef.current.value = ''
+            newInstrumentInputRef.current.focus()
+          }}
+        >
+          <i className="fas fa-plus-circle" style={{ background: '0' }}></i>
+        </AddNewMembersInstrument>
+      </Row>
+      <RowFlexStart>
+        {newMembersInstruments.map(instrument => (
+          <Tag
+            key={instrument}
+            onClick={event => {
+              event.preventDefault()
+              removeNewMembersInstrument(instrument)
+            }}
+          >
+            {instrument}
+          </Tag>
+        ))}
+      </RowFlexStart>
+
+      <Submit type="submit">Create new member</Submit>
+    </Form>
+  )
+
+  function createMember(event) {
+    const formData = event.target
+    event.preventDefault()
+
+    const newId = validateId()
+    const roles = collectRoles()
+    const band = ascertainBand()
+
+    const newMember = {
+      id: newId,
+      name: formData.memberName.value,
+      band: band,
+      roles: roles,
+      instruments: newMembersInstruments,
+      roomAsignment: 'RehearsalRoom_1',
+      repetitionDayTime: [],
+      avatar: '',
+    }
+    postMember(newMember).then(() => {
+      setMembers([...members, newMember])
+      resetNewMembersInstruments()
+      toggleForm()
+    })
+
+    function collectRoles() {
+      const rawRoles = [
+        formData.administrator.checked
+          ? formData.administrator.value
+          : !formData.administrator,
+        formData.musician.checked
+          ? formData.musician.value
+          : !formData.musician,
+        !formData.administrator.checked && !formData.musician.checked
+          ? 'guest'
+          : false,
+      ]
+      return rawRoles.filter(Boolean)
+    }
+
+    function ascertainBand() {
+      const bandIs = formData.band.value
+        ? formData.band.value
+        : formData.newBand.value
+        ? formData.newBand.value
+        : null
+      return bandIs
+    }
+
+    // The developer is open to speak about more elagant solution for such validation. Code-reviewers welcomed!
+    function validateId() {
```

Comment:
There is not need to generate user ids this way.. simply use the uid package be be done with it. The more users your app has the more recursive calls and this will definitely break once you reach the 1000 user mark!

Reply:
Hardly there will be more than 1000 members in a single rehearsal room. normally there are two dozens maximally. Even with couple hundreeds guests, there still will be place. I will think about scalling when I will make a log-in.